guides/source/working_with_javascript_in_rails.md

```diff
@@ -3,561 +3,212 @@
 
-[`button_to`](https://api.rubyonrails.org/classes/ActionView/Helpers/UrlHelper.html#method-i-button_to) is a helper that helps you create buttons. It has a `:remote` option that you can call like this:
-
-```erb
-<%= button_to "An article", @article, remote: true %>
+```bash
+node --version
 ```
 
-this generates
-
-```html
-<form action="/articles/1" class="button_to" data-remote="true" method="post">
-  <input type="submit" value="An article" />
-</form>
-```
-
-Since it's just a `<form>`, all the information on `form_with` also applies.
-
-### Customize Remote Elements
-
-It is possible to customize the behavior of elements with a `data-remote`
-attribute without writing a line of JavaScript. You can specify extra `data-`
-attributes to accomplish this.
-
-#### `data-method`
+The version of your Node.js runtime should be printed out. Make sure it’s greater than 8.16.0.
 
-Activating hyperlinks always results in an HTTP GET request. However, if your
-application is [RESTful](https://en.wikipedia.org/wiki/Representational_State_Transfer),
-some links are in fact actions that change data on the server, and must be
-performed with non-GET requests. This attribute allows marking up such links
-with an explicit method such as "post", "put" or "delete".
+To install Yarn, follow the installation instructions at the  [Yarn website](https://classic.yarnpkg.com/en/docs/install) .
+Running this command should print out Yarn version:
 
-The way it works is that, when the link is activated, it constructs a hidden form
-in the document with the "action" attribute corresponding to "href" value of the
-link, and the method corresponding to `data-method` value, and submits that form.
-
-NOTE: Because submitting forms with HTTP methods other than GET and POST isn't
-widely supported across browsers, all other HTTP methods are actually sent over
-POST with the intended method indicated in the `_method` parameter. Rails
-automatically detects and compensates for this.
-
-#### `data-url` and `data-params`
-
-Certain elements of your page aren't actually referring to any URL, but you may want
-them to trigger Ajax calls. Specifying the `data-url` attribute along with
-the `data-remote` one will trigger an Ajax call to the given URL. You can also
-specify extra parameters through the `data-params` attribute.
-
-This can be useful to trigger an action on check-boxes for instance:
-
-```html
-<input type="checkbox" data-remote="true"
-    data-url="/update" data-params="id=10" data-method="put">
+```bash
+yarn --version
 ```
 
-#### `data-type`
-
-It is also possible to define the Ajax `dataType` explicitly while performing
-requests for `data-remote` elements, by way of the `data-type` attribute.
+If it says something like 1.22.0, Yarn has been installed correctly.
 
-### Confirmations
+Choosing Between Import Maps and a JavaScript Bundler
```

Comment:
@rubys @ghiculescu Here's a first cut at addressing the conversation about which to choose and the challenges of migrating. Very much open to improvements on this section, I'm not confident I've properly captured the why of import maps well enough. There may also be room for improvement on the encouragement to carefully consider the choice.

Reply:
I really like this!